glean-core/src/common_metric_data.rs

```diff
@@ -81,18 +86,36 @@ impl CommonMetricData {
         }
     }
 
-    /// The metric's unique identifier.
+    /// The metric's base identifier, including the category and name, but not the label.
     ///
-    /// If `category` is empty, it's just the name.
+    /// If `category` is empty, it's ommitted.
     /// Otherwise, it's the combination of the metric's `category` and `name`.
-    pub fn identifier(&self) -> String {
+    pub fn base_identifier(&self) -> String {
         if self.category.is_empty() {
             self.name.clone()
         } else {
             format!("{}.{}", self.category, self.name)
         }
     }
 
+    /// The metric's unique identifier, including the category, name and label.
+    ///
+    /// If `category` is empty, it's ommitted.
+    /// Otherwise, it's the combination of the metric's `category`, `name` and `label`.
+    pub fn identifier(&self, glean: &Glean) -> String {
```

Comment:
Instead of changing this to require `glean` for labels, given that it has repercussions on the whole codebase, what about introducing a new `identifier_with_label` and only require the Glean object for that function?

Reply:
I thought about this. Strictly speaking, you only need to call this version on metrics that can be labeled.  But I thought that would just introduce confusion about when one should be used over another and lead to unexpected bugs...  I'm happy to change this, though, if you feel strongly.

Reply:
Fine by me then, with the changes by @badboy